pkg/apis/batch/validation/validation.go

```diff
@@ -564,13 +568,16 @@ func validateConcurrencyPolicy(concurrencyPolicy *batch.ConcurrencyPolicy, fldPa
 	return allErrs
 }
 
-func validateScheduleFormat(schedule string, timeZone *string, fldPath *field.Path) field.ErrorList {
+func validateScheduleFormat(schedule, oldSchedule string, timeZone *string, fldPath *field.Path) field.ErrorList {
```

Comment:
rather than passing in the whole schedule, I think it would be easier to understand and unit test to make the parameter here as narrow as possible, like `allowTZInSchedule bool`

could invoke it like this:
```go
allowTZInSchedule := false
if oldSpec != nil {
  allowTZInSchedule = strings.Contains(oldSpec.Schedule, "TZ")
}
allErrs = append(allErrs, validateScheduleFormat(spec.Schedule, allowTZInSchedule, spec.TimeZone, fldPath.Child("schedule"))...)
```

and use it like this:
```go
switch {
case allowTZInSchedule && strings.Contains(schedule, "TZ") && timeZone != nil:
  allErrs = append(allErrs, field.Invalid(fldPath, schedule, "cannot use both timeZone field and TZ or CRON_TZ in schedule"))
case !allowTZInSchedule && strings.Contains(schedule, "TZ"):
  allErrs = append(allErrs, field.Invalid(fldPath, schedule, "cannot use TZ or CRON_TZ in schedule, use timeZone field instead"))
}
```

Reply:
Great suggestion - updated.